internal/checker/relater.go

```diff
@@ -3365,15 +3365,8 @@ func (r *Relater) structuredTypeRelatedToWorker(source *Type, target *Type, repo
 		if r.relation == r.c.comparableRelation && source.flags&TypeFlagsTypeParameter != 0 {
 			// This is a carve-out in comparability to essentially forbid comparing a type parameter with another type parameter
 			// unless one extends the other. (Remember: comparability is mostly bidirectional!)
-			constraint := r.c.getConstraintOfTypeParameter(source)
-			if constraint != nil {
-				for constraint != nil && someType(constraint, func(c *Type) bool { return c.flags&TypeFlagsTypeParameter != 0 }) {
-					result = r.isRelatedTo(constraint, target, RecursionFlagsSource, false /*reportErrors*/)
-					if result != TernaryFalse {
-						return result
-					}
-					constraint = r.c.getConstraintOfTypeParameter(constraint)
-				}
```

Comment:
The loop here (a) isn't needed and (b) potentially treats a union type object as a type parameter object. Since `isRelatedTo` recurses back to here for each level of constraints, indirect constraints (such as `C extends B` and `B extends A`) need no extra handling. And, by getting rid of the loop, we also get rid of the potential panic from the `getConstraintOfTypeParameter` call in the loop.